doc/Data Plane Code Generation/DeveloperDrivenEvolution.md

```diff
@@ -0,0 +1,195 @@
+# Developer driven evolution
+To add a convenience API that takes and returns strongly-typed input and output models instead of raw JSON, steps are:
+1. [Create the convenience method with new signature](#create-the-convenience-method-with-new-signature)
+- [Replace input parameters](#replace-input-parameters)
+- [Replace output value](#replace-output-value)
+- [Rename method if needed](#rename-method-if-needed)
+2. [Create the models](#create-the-models)
+3. [Implement the method](#implement-the-method)
+- [Implement GET method](#implement-get-method)
+- [Implement POST method](#implement-post-method)
+- [Implement GET paging method](#implement-get-paging-method)
+
+## Create the convenience method with new signature
+You should first decide how the convenience method grows up from protocol method. Below is an example of your generated protocol method.
+```C#
+public virtual async Task<Response> CreateMetricFeedbackAsync(RequestContent content, RequestContext context = null);
+```
+You should do the following steps to create the convenience method signature:
```

Comment:
Could you explain more on this?  Are we expecting developer to modify the existing method or to create a new method with the same name but different input/output?

Reply:
They should not modify generated code.  We use partial classes to add grow-up methods in a separate file that will not be overwritten by the generator.  This information should be added so partners know where to add the grow-up methods.